### DISCUSSION TITLE
Is it possible to get `event.target.selectionStart` from onClick event

### DISCUSSION THREAD

**Discussion by mcajic** [remarkjs/remark#1062] [Q&A] [answered]
I have a text input whose content is displayed under `<ReactMarkdown />` component after editing. I handle click on markdown wrapper element to hide markdown and display text input to edit backing up text. Is it possible to get position of clicked text in `<ReactMarkdown />` (similar to event.target.selectionStart on the input element) so I can focus the same position in the text input?

```
editing ? (
    <UserPropertyTextFieldStyled
      onBlur={save}
      placeholder="empty"
      value={currentValue}
      onChange={handleChange}
      onClick={handleClick}
      ref={ref}
    />
  ) : (
    <MarkdownBoxStyled onClick={() => setEditing(true)}>
      <ReactMarkdown>{currentValue}</ReactMarkdown>
    </MarkdownBoxStyled>
  )
```

**Comment by ChristianMurphy** [comment, marked as answer]
> Is it possible to get position of clicked text

It is possible.
However doing the source mapping will not necessarily be easy.
Some ideas for getting started.
* Consider using the `components` prop, to wrap any/all of the elements you want to track clicks for, and add your event listeners to the wrapper elements
* Consider adding native DOM effect, doing a `querySelectorAll` on the content inside `<ReactMarkdown>` and adding event listeners to each, then using that to estimate position.
* Consider using a pre-built editor like:
   * https://github.com/hanford/remark-slate
   * https://github.com/inokawa/remark-slate-transformer
   * likely more, try looking through: https://github.com/topics/remark-plugin
